michelson.md

```diff
@@ -1073,6 +1073,8 @@ The `#DoCompare` function requires additional lemmas for symbolic execution.
   rule X ==String X => true  [simplification]
   rule X  <String X => false [simplification]
 
+  rule #Ceil(#DoCompare(_:Int, _:Int)) => #Top [anywhere, simplification]
+
   rule #Ceil(#DoCompare(V1, V2)) => #Top
     requires (isBool(V1)      andBool isBool(V2))
```

Comment:
Should we split on every sort like this?

Reply:
Do we have reason to suspect that will lead to an improvement? I'm okay with leaving it as it is.

Also, how would we split on every sort? I think I've only seen `#DoCompare` applied to `Int`s in our proofs so far.